src/System.Data.SqlClient/tests/ManualTests/SQL/MirroringTest/ConnectionOnMirroringTest.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.Data.SqlClient.ManualTesting.Tests
+{
+    public static class ConnectionOnMirroringTest
+    {
+        [CheckConnStrSetupFact]
+        public static void TestMultipleConnectionToMirroredServer()
+        {
+            string connectionString = DataTestUtility.TcpConnStr;
+            int mirroringState;
+            string failoverPartnerName;
+            bool isMirroring = GetMirroringInfo(connectionString, out mirroringState, out failoverPartnerName);
+            List<SqlConnection> list = new List<SqlConnection>();
+            if (isMirroring && mirroringState == 4 && !string.IsNullOrEmpty(failoverPartnerName))
+            {
+                Stopwatch stopWatch = new Stopwatch();
+                TestWorker worker = new TestWorker(connectionString);
+                Thread childThread = new Thread(() => worker.TestMultipleConnection());
+
+                stopWatch.Start();
+                childThread.Start();
+                while (!worker.IsDone && stopWatch.ElapsedMilliseconds <= 10000);
```

Comment:
Why not use Thread.Sleep() for a few milliseconds if the worker hasn't completed?

Reply:
Because child thread could finish its job earlier than the number of milliseconds the main thread sleeps for. Main thread does not need to wait extra time after child finishes.

Reply:
10000 ms wait is arbitrary. shouldn't we wait for the connection timeout period?

Reply:
How about using ManualResetEvent to signal when the thread completes, and then waiting on that for 10 seconds instead? 
https://msdn.microsoft.com/en-us/library/system.threading.manualresetevent(v=vs.110).aspx

Reply:
@saurabh500 the connection timeout for testing the scenario of the bug (https://github.com/dotnet/corefx/issues/24148) is 0.

Reply:
@corivera Is this what you meant?
* Main thread is blocked by `ManualResetEvent.Reset()` after starting child thread.
* Child thread completes its task, and call `ManualResetEvent.Set()`
* Main thread is unblocked, and finishes the application.

Reply:
Reset() doesn't block the thread, it just sets the Event state to nonsignaled. You need to call WaitOne() on the Event to actually wait on it. Instead of setting IsDone=true, you just call ManualResetEvent.Set(), which will unblock waiting thread.

Also, where are you setting timeout=0 in the test? The default connection timeout is 15 seconds.

Reply:
But it does the same thing anyway. The only difference is the main thread spins until the child thread finishes or be timed out.
And, this is manual test. connection string is provided manually by user. I will put a line to force to set timeout 0.

Reply:
Up to you. It just seems like ManualResetEvent signaling does exactly what you want without needing any of the busy spinning (for up to 10 seconds) in between.